### PR TITLE
Implemented vsphere network changing

### PR DESCRIPTION
**Pull request by pkavajin**
**What this PR does / why we need it**:
Adds support for changing the vm network when machine controller creates a new vm

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #239

**Special notes for your reviewer**:
Do not merge. godep versions have to be updated as soon as corresponding machine-controller pr was merged.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/api/v2/node.go [hunk 1]
```diff
@@ -119,9 +119,10 @@ type AzureNodeSpec struct {
 // VSphereNodeSpec VSphere node settings
 // swagger:model VSphereNodeSpecV2
 type VSphereNodeSpec struct {
-	CPUs     int    `json:"cpus"`
-	Memory   int    `json:"memory"`
-	Template string `json:"template"`
+	CPUs            int    `json:"cpus"`
+	Memory          int    `json:"memory"`
+	Template        string `json:"template"`
+	TemplateNetName string `json:"templateNetName"`
```
Comment (alvaroaleman) [inline api/pkg/api/v2/node.go:125]: We need both the `TemplateNetName` and the `VMNetName` here as the latter could be different for different templates

File: api/pkg/machine/convert.go [hunk 2]
```diff
@@ -126,9 +126,10 @@ func GetAPIV2NodeCloudSpec(machine *v1alpha1.Machine) (*apiv2.NodeCloudSpec, err
 			return nil, fmt.Errorf("failed to parse hetzner config: %v", err)
 		}
 		cloudSpec.VSphere = &apiv2.VSphereNodeSpec{
-			CPUs:     int(config.CPUs),
-			Memory:   int(config.MemoryMB),
-			Template: config.TemplateVMName.Value,
+			CPUs:            int(config.CPUs),
+			Memory:          int(config.MemoryMB),
+			Template:        config.TemplateVMName.Value,
+			TemplateNetName: config.TemplateNetName.Value,
```
Comment (alvaroaleman) [inline api/pkg/machine/convert.go:132]: So we assume Vms always have exactly one network adapter?

**Review by alvaroaleman — CHANGES_REQUESTED** [review]

**Comment by alvaroaleman** [comment]
Also, please execute `./hack/update-swagger.sh` to update the swagger spec

File: api/pkg/machine/convert.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/machine/convert.go:132]: Yes. For now a VM only has one adapter. That's the default in all OVA-templates (Ubuntu/Coreos/Centos).
If a customer want's to configure more, the template needs to be customized

**Review by alvaroaleman — DISMISSED** [review]

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**